### PR TITLE
DOC: How to use storage_option param here to integrate with GCP bucket? 

### LINKED ISSUE #42066: DOC: How to use storage_option param here to integrate with GCP bucket?
#### Location of the documentation

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_pickle.html

#### Documentation problem

How to use storage_option param here to integrate with GCP bucket?

#### Suggested fix for documentation

Please add some examples to pandas.DataFrame.to_{any_file_format}.html

### PR DESCRIPTION
**Pull request by purna135**
- [x] closes #42066 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by twoertwein** [comment]
> Hi @MarcoGorelli, I altered a few files, but there are still a few that offer a brief description of the storage options, as shown below. Please let me know if these files need to be corrected as well.

If possible, it might be great to share the doc-strings between all of these methods. I think we already have shared docs for `storage_options`:

```py
from pandas.util._decorators import doc
from pandas.core.shared_docs import _shared_docs

@doc(storage_options=_shared_docs["storage_options"])
def some_function():
    """
    {storage_options}
    """
```

**Comment by twoertwein** [comment]
> Hello, @twoertwein. Could you please assist me in determining what I am doing incorrectly?

What are you having trouble with? Some tests sometimes randomly fail, the faillures do not seem related to your changes.

**Comment by purna135** [comment]
> What are you having trouble with? Some tests sometimes randomly fail, the faillures do not seem related to your changes.

Thank you : )
I had previously added `{storage options}` to the base file, but it failed several tests, so I redo the changes. 
let me change and push it again.

File: pandas/core/shared_docs.py [hunk 1]
```diff
@@ -402,7 +402,10 @@
     host, port, username, password, etc. For HTTP(S) URLs the key-value pairs
     are forwarded to ``urllib`` as header options. For other URLs (e.g.
     starting with "s3://", and "gcs://") the key-value pairs are forwarded to
-    ``fsspec``. Please see ``fsspec`` and ``urllib`` for more details."""
+    ``fsspec``. Please see ``fsspec`` and ``urllib`` for more details,
```
Comment (twoertwein) [inline pandas/core/shared_docs.py:405]: Not sure whether the following  is needed: could even be more concrete by writing something like

> For HTTP(S) URLs the key-value pairs are forwarded to ``urllib.request.Request`` as header options. For other URLs (e.g. starting with "s3://", and "gcs://") the key-value pairs are forwarded to ``fsspec.open``

File: pandas/io/excel/_base.py [hunk 2]
```diff
@@ -272,13 +272,7 @@
...
     'X'...'X'. Passing in False will cause data to be overwritten if there
     are duplicate names in the columns.
-storage_options : dict, optional
-    Extra options that make sense for a particular storage connection, e.g.
-    host, port, username, password, etc., if using a URL that will
-    be parsed by ``fsspec``, e.g., starting "s3://", "gcs://". An error
-    will be raised if providing this argument with a local path or
-    a file-like buffer. See the fsspec and backend storage implementation
-    docs for the set of allowed keys and values.
+{storage_options}
```
Comment (twoertwein) [inline pandas/io/excel/_base.py:275]: This string does contain many `{}`s, I think you will need to replace all `{...}` with `{{...}}` (unless they are intended to be replaced).

File: pandas/core/shared_docs.py (hunk 1 above)
Comment (purna135) [reply pandas/core/shared_docs.py:405]: Do you mean, I can replace the storage_options as bellow ?

storage_options : dict, optional
    Extra options that make sense for a particular storage connection, e.g. host, port, username, password, etc. For HTTP(S) URLs the key-value pairs are forwarded to ``urllib.request.Request`` as header options. For other URLs (e.g. starting with "s3://", and "gcs://") the key-value pairs are forwarded to ``fsspec.open``. Please see ``fsspec`` and ``urllib`` for more details, and for more examples on storage options refer [here](https://pandas.pydata.org/docs/user_guide/io.html?highlight=storage_options#reading-writing-remote-files)

Comment (twoertwein) [reply pandas/core/shared_docs.py:405]: Yes, looks good to me.

Comment (purna135) [reply pandas/core/shared_docs.py:405]: Thanks, I will commit with these changes

**Comment by MarcoGorelli** [comment]
Hey @purna135 - did you try building this locally? I think it'd be helpful if you could post screenshots of the amended docstrings

**Comment by purna135** [comment]
Yes I have built this locally and bellow is the screanshot of amended docstrings.

image

**Review by twoertwein — APPROVED** [review]
Thanks @purna135! Let's wait for the CI to finish but otherwise looks good to me!

**Comment by purna135** [comment]
Should I share a screencap of the part where I replaced `{...}` with `{{...}` to repair the unescaped braces error?

**Review by MarcoGorelli — APPROVED** [review]
Thanks @purna135 , and thanks @twoertwein for reviewing!

**Pull request merged**